### PR TITLE
ICU-22205 toolutil: Fix MASM generation for x86-64 and ARM64

### PR DESCRIPTION
**Pull request by amyspark**
👋

This is the associated PR to fix bug [ICU-22205](https://unicode-org.atlassian.net/browse/ICU-22205).

##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22205
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

[ICU-22205]: https://unicode-org.atlassian.net/browse/ICU-22205?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: icu4c/source/tools/toolutil/pkg_genc.cpp [hunk 1]
```diff
@@ -228,6 +228,17 @@ static const struct AssemblyType {
     },
 /* align 16 bytes */
 /*  http://msdn.microsoft.com/en-us/library/dwa9fwef.aspx */
+#if defined(_WIN64)
```
Comment (markusicu) [inline icu4c/source/tools/toolutil/pkg_genc.cpp:231]: I propose that we don't use `#if` but instead give this a different name like "masm64". (Or "masm", and rename the old one to "masm32".) That would let us generate either one on any machine.

**Comment by amyspark** [comment]
@markusicu I've added the corresponding change of extension. The Meson build system uses `masm` for MASM files, but everyone else uses `asm` so I've kept the latter.

**Comment by amyspark** [comment]
@markusicu The CI failures look unrelated, can I safely ignore them?

**Comment by amyspark** [comment]
@markusicu Ping for review, let me know if there's anything needed here.

**Review by srl295 — APPROVED** [review]
looks plausible to me

**Review by markusicu — APPROVED** [review]
yes, looks plausible; sorry for the long delay...

**Pull request merged**